recipe_scrapers/joshuaweissman.py

```diff
@@ -0,0 +1,70 @@
+# mypy: allow-untyped-defs
+
+# from pprint import pprint
+from ._abstract import AbstractScraper
+from ._grouping_utils import group_ingredients
+from ._utils import normalize_string
+
+
+class JoshuaWeissman(AbstractScraper):
+    @classmethod
+    def host(cls):
+        return "joshuaweissman.com"
+
+    def author(self):
+        return self.schema.author()
+
+    def title(self):
+        return self.schema.title()
+
+    def category(self):
+        return self.schema.category()
+
+    def total_time(self):
+        raise NotImplementedError()
+
+    def yields(self):
+        raise NotImplementedError()
+
+    def image(self):
+        return self.schema.image()
+
+    def ingredients(self):
+        ingredients = [
+            li.get_text() for ul in self.soup.findAll("ul") for li in ul.findAll("li")
+        ]
+
+        ingredients = [
+            normalize_string(ingredient) for ingredient in ingredients if ingredient
+        ]
+
+        return ingredients
+
+    def ingredient_groups(self):
+        # This is the css for it but web site has weird structuring so it doesn't work properly
+        return group_ingredients(
+            self.ingredients(),
+            self.soup,
+            'span[style="font-weight: 700;"]',
+            "ul>li",
+        )
+
+    def instructions(self):
+        instructions = [
+            li.get_text() for ol in self.soup.findAll("ol") for li in ol.findAll("li")
+        ]
+
+        instructions = [
+            normalize_string(instruction) for instruction in instructions if instruction
+        ]
+
+        return "\n".join(instructions)
+
+    def ratings(self):
+        return self.schema.ratings()
+
+    def cuisine(self):
+        return self.schema.cuisine()
+
+    def description(self):
+        return self.soup.find(id="viewer-foo").get_text()
```

Comment:
Hrm.  It seems weird that the `schema.org` description and `og:description` field contain other data -- and maybe this selector is a bit fragile.  But the output does seem better, so this seems looks like the way to go.

Reply:
Yeah, I don't really know what they are doing with or how the schema is generated, cause on some of them it would start putting the ingredients into the schema, which was mainly why I ended up with that as the selector.

Somewhat also similar to that, this site is for a [youtube channel](https://www.youtube.com/@JoshuaWeissman), which as far as I understand all of the recipes were made by the Joshua dude. But the author fields in the schema are for other people, so I debated on changing that to hard code "Joshua Weissman" but decided to not hard code it and leave it as what the schema is saying, and the site can change it later themselves if they want.

Overall, I think, as mentioned in some other issues I saw, Wix is not incredibly friendly to parse.